### PR TITLE
Grafana-UI: update date picker

### LINKED ISSUE #22411: Grafana UI: Update TimeOfDayPicker to use new form styles
**What would you like to be added**:
Styling for `TimeOfDayPicker` should be updated to be consistent with `Forms` component styles. 

**Why is this needed**:
This would enable using `TimeOfDayPicker` together with new form components.

### PR DESCRIPTION
**Pull request by Clarity-89**
**What this PR does / why we need it**:
Apply new form styles to the `TimeOfDayPicker`.
**Which issue(s) this PR fixes**:

Fixes #22411 

Before:
Screenshot 2020-02-25 at 11 58 25
Screenshot 2020-02-25 at 11 58 31

After: 
Screenshot 2020-02-25 at 11 40 52
Screenshot 2020-02-25 at 11 59 19

### CODE REVIEW

File: packages/grafana-ui/src/components/TimePicker/TimeOfDayPicker.tsx [hunk 1]
```diff
@@ -16,6 +16,11 @@ interface Props {
 }
 
 const getStyles = stylesFactory((theme: GrafanaTheme) => {
+  const bgColor = stv({ light: theme.colors.white, dark: theme.colors.gray15 }, theme.type);
```
Comment (torkelo) [inline packages/grafana-ui/src/components/TimePicker/TimeOfDayPicker.tsx]: should this not be formInputBg

**Review by torkelo — APPROVED** [review]
Looks good just one theme variable use could be added

File: packages/grafana-ui/src/components/TimePicker/TimeOfDayPicker.tsx (hunk 1 above)
Comment (Clarity-89) [reply packages/grafana-ui/src/components/TimePicker/TimeOfDayPicker.tsx]: Ah right, didn't know `gray15` had an alias.

**Review by dprokop — CHANGES_REQUESTED** [review]
image
I can see some visual glitches to be fixed. Not a huge fan of the 3D effect to divide the columns, I don't think we use this effect anywhere in Grafana.

**Comment by Clarity-89** [comment]
@dprokop good catch! There was an extra box-shadow on the internal panel, after removing it the picker looks like this:
Screenshot 2020-02-25 at 19 45 01

**Comment by dprokop** [comment]
Also mind the border of the "input" that's rendered:

image

**Comment by Clarity-89** [comment]
@dprokop I think that's intentional since the time selector is an overlay on top of the input element, and not a dropdown as is the case with Select components. At least that's my impression by looking at their [examples](https://react-component.github.io/time-picker/examples). But I can tweak it a bit so it's not so prominent.

**Comment by Clarity-89** [comment]
The tweaked version: 
Screenshot 2020-02-25 at 20 52 50

**Comment by dprokop** [comment]
Ok, I see. But this still looks off, see the overly element covering the focus fing. Any chance we can make the overly the same width as the underlying input and make it have top/left corner at the same position/

**Comment by Clarity-89** [comment]
@dprokop do you mean smth like this? 
Screenshot 2020-02-26 at 9 29 11

**Comment by dprokop** [comment]
Exactly! Probably the border radius needs to be adjusted and make sure the positioning of the element is aligned with the original input.

**Comment by Clarity-89** [comment]
@dprokop adjusted borders

Screenshot 2020-02-26 at 10 30 29

**Comment by dprokop** [comment]
Looks way better now. Just one last thing that can be seen here: https://youtu.be/i9z4eeRgZ7o (use 0.2 speed). When the dropdown is shown it's 1px wider on each side than the original input making it look like the input is jumping

**Comment by Clarity-89** [comment]
@dprokop I've aligned the overlay size with input and now there's no jumping effect. The value does shift a tiny bit to the left, but it's not obvious and the shift seems to be less than a pixel so I wasn't able to completely remove it.

**Comment by dprokop** [comment]
AH, there's always the "one last thing...", sorry:

image
 
is the blue line around selected hour/minute by design?

**Comment by Clarity-89** [comment]
The outline is by default and I left it for accessibility purposes - you can select hour/minute by using tab key and pressing enter. Without that it's impossible to know which element is selected.

**Comment by dprokop** [comment]
I see, good stuff then!

**Review by dprokop — APPROVED** [review]

**Pull request merged**